internal/conditions/types/spiffe.go

```diff
@@ -0,0 +1,566 @@
+	overloadSpiffeIDPath           = "path"
+	overloadSpiffeIDTrustDomain    = "trustDomain"
+	overloadSpiffeMatcherMatchesID = "matchesID"
+	overloadSpiffeTrustDomainID    = "id"
+	overloadSpiffeTrustDomainName  = "name"
+)
+
+var (
+	SPIFFEIDType          = cel.ObjectType(spiffeIDTypeName, traits.ReceiverType, traits.ComparerType)
+	SPIFFETrustDomainType = cel.ObjectType(spiffeTrustDomainTypeName, traits.ReceiverType, traits.ComparerType)
+	SPIFFEMatcherType     = cel.ObjectType(spiffeMatcherTypeName, traits.ReceiverType)
+
+	spiffeIDCelType = cel.ObjectType(spiffeIDTypeName)
+
+	SPIFFEIDFunc = cel.Function(spiffeIDFn,
+		cel.Overload(
+			fmt.Sprintf("%s_string", spiffeIDFn),
+			[]*cel.Type{cel.StringType},
+			spiffeIDCelType,
+			cel.UnaryBinding(unarySPIFFEIDFnImpl),
+		),
+	)
+
+	spiffeTrustDomainCelType = cel.ObjectType(spiffeTrustDomainTypeName)
+
+	SPIFFETrustDomainFunc = cel.Function(spiffeTrustDomainFn,
+		cel.Overload(
+			fmt.Sprintf("%s_string", spiffeTrustDomainFn),
+			[]*cel.Type{cel.StringType},
+			spiffeTrustDomainCelType,
+			cel.UnaryBinding(unarySPIFFETrustDomainFnImpl),
+		),
+	)
+
+	spiffeMatcherCelType = cel.ObjectType(spiffeMatcherTypeName)
+
+	SPIFFEMatchAnyFunc = cel.Function(spiffeMatchAnyFn,
+		cel.Overload(
+			spiffeMatchAnyFn,
+			nil,
+			spiffeMatcherCelType,
+			cel.FunctionBinding(unarySPIFFEMatchAnyFnImpl),
+		),
+	)
+
+	SPIFFEMatchExactFunc = cel.Function(spiffeMatchExactFn,
+		cel.Overload(
+			fmt.Sprintf("%s_spiffeID", spiffeMatchExactFn),
+			[]*cel.Type{spiffeIDCelType},
+			spiffeMatcherCelType,
+			cel.UnaryBinding(unarySPIFFEMatchExactFnImpl),
+		),
+
+		cel.Overload(
+			fmt.Sprintf("%s_string", spiffeMatchExactFn),
+			[]*cel.Type{cel.StringType},
+			spiffeMatcherCelType,
+			cel.UnaryBinding(unarySPIFFEMatchExactFnImpl),
+		),
+	)
+
+	SPIFFEMatchOneOfFunc = cel.Function(spiffeMatchOneOfFn,
+		cel.Overload(
+			fmt.Sprintf("%s_spiffeIDList", spiffeMatchOneOfFn),
+			[]*cel.Type{cel.ListType(spiffeIDCelType)},
+			spiffeMatcherCelType,
+			cel.UnaryBinding(unarySPIFFEMatchOneOfFnImpl),
+		),
+
+		cel.Overload(
+			fmt.Sprintf("%s_stringList", spiffeMatchOneOfFn),
+			[]*cel.Type{cel.ListType(cel.StringType)},
+			spiffeMatcherCelType,
+			cel.UnaryBinding(unarySPIFFEMatchOneOfFnImpl),
+		),
+	)
+
+	SPIFFEMatchTrustDomainFunc = cel.Function(spiffeMatchTrustDomainFn,
+		cel.Overload(
+			fmt.Sprintf("%s_spiffeTrustDomain", spiffeMatchTrustDomainFn),
+			[]*cel.Type{spiffeTrustDomainCelType},
+			spiffeMatcherCelType,
+			cel.UnaryBinding(unarySPIFFEMatchTrustDomainFnImpl),
+		),
+
+		cel.Overload(
+			fmt.Sprintf("%s_string", spiffeMatchTrustDomainFn),
+			[]*cel.Type{cel.StringType},
+			spiffeMatcherCelType,
+			cel.UnaryBinding(unarySPIFFEMatchTrustDomainFnImpl),
+		),
+	)
+
+	spiffeIDTypeExpr          = types.NewObjectType(spiffeIDTypeName)
+	spiffeMatcherTypeExpr     = types.NewObjectType(spiffeMatcherTypeName)
+	spiffeTrustDomainTypeExpr = types.NewObjectType(spiffeTrustDomainTypeName)
+
```

Comment:
Are these type declarations needed?  They are duplicates of the types defined earlier in this file:
```
	SPIFFEIDType          = cel.ObjectType(spiffeIDTypeName, traits.ReceiverType, traits.ComparerType)
	SPIFFETrustDomainType = cel.ObjectType(spiffeTrustDomainTypeName, traits.ReceiverType, traits.ComparerType)
	SPIFFEMatcherType     = cel.ObjectType(spiffeMatcherTypeName, traits.ReceiverType)
```

`cel.ObjectType` is an alias of `types.NewObjectType`.